client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/TopologyInfo.java

```diff
@@ -39,11 +39,11 @@
    private Collection<SocketAddress> servers = new ArrayList<>();
    private Map<byte[], ConsistentHash> consistentHashes = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
    private Map<byte[], Integer> segmentsByCache = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
-   private volatile AtomicInteger topologyId;
+   private Map<byte[], AtomicInteger> topologyIds = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
```

Comment:
Ok, I asked because ```TopologyInfo``` as you can see is not thread-safe, and is always supposed to be called  under a lock in ```TcpTransportFactory```